source/hwPortUtils.py

```diff
@@ -127,147 +131,115 @@ def _isDebug():
 
 
 # C901 'listComPorts' is too complex. Look for opportunities to break this method down.
-def listComPorts(onlyAvailable=True) -> typing.Iterator[typing.Dict]:  # noqa: C901
+def listComPorts(onlyAvailable: bool = True) -> typing.Iterator[dict]:  # noqa: C901
 	"""List com ports on the system.

```

Comment:
considering this is a full rewrite, the complexity should be reduced to remove the c901. I'd encourage splitting this up into helper functions